### PR TITLE
Auth tvOS build environment + email/password example.

### PR DESCRIPTION
**Pull request by ryanwilson**

### CODE REVIEW

File: Firebase/Auth/Source/FIRAuthAPNSTokenManager.m [hunk 1]
```diff
@@ -73,7 +73,9 @@ - (void)getTokenWithCallback:(FIRAuthAPNSTokenCallback)callback {
     } else {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#if !TARGET_OS_TV
```
Comment (paulb777) [inline Firebase/Auth/Source/FIRAuthAPNSTokenManager.m]: Does the style guide cover #if !?  

In some places, #if !defined(TARGET_OS_TV) is preferred.

Also, uglier, but more correct, to also #if the if respondsToSelector ...

Comment (ryanwilson) [reply Firebase/Auth/Source/FIRAuthAPNSTokenManager.m]: I didn't see it but happy to do `!defined` instead. Will update it!

Comment (ryanwilson) [reply Firebase/Auth/Source/FIRAuthAPNSTokenManager.m]: Re: the `respondsToSelector`, tvOS actually does support the above call, just not this one 😄

Comment (paulb777) [reply Firebase/Auth/Source/FIRAuthAPNSTokenManager.m]: Understood - so the if will always be true on tvOS, but if it wasn't nothing would happen in the else.

If this was pure tvOS code, there would be no reason for the respondsToSelector check, but I'm fine with keeping the code easier to read.

**Pull request merged**